tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -623,7 +623,7 @@ describe("updateTechRecord", () => {
           expect(await techRecordsService.updateTechRecord(recordToUpdate, msUserDetails, ["nsa7zXuM/5iYmrCM2kzmT"])).toThrowError();
         } catch (errorResponse) {
           expect(errorResponse).toBeInstanceOf(HTTPError);
-          expect(errorResponse.statusCode).toEqual(500);
+          expect(errorResponse.statusCode).toEqual(404);
```

Comment:
This would probably qualify as a 500 if the client can't change anything to fix it. Makes sense that AWS would return 404, but that's not relevant to our users/the app

Reply:
The thing is, any error related to upload will be caught by the catch block in TechRecordService line 193. "uploadFile" function returns a promise from AWS that's appended in that promises array, but we don't know if it's a reject or a resolve...That catch block will also receive any errors from manageUpdateLogic, which can be 400 Validation error, 404 Not Found, Conditional Request failed, 422 More than one match, etc. Having said this, the old implementation was wrong, because I was returning error 500 and assuming that was an AWS error, when that was not the case. It can be any of the above, that's why on that catch block (line 193 TechRecService), instead of returning error code 500 (hardcoded) I'm now returning whatever error code is thrown. And that's why in this test I'm checking for a 404, because that's what's returned by AWS. Hope that makes sense :)

Reply:
done, returns 500 now